### DISCUSSION TITLE
Question about state.json

### DISCUSSION THREAD

**Discussion by ghost** [Koenkk/zigbee2mqtt#19468] [General]
Hi everyone,

I have a quick question (actually 2) about the state.json file.

So, when does the state.json gets updated? I sometimes experience that the state of my devices are different to the state.json file.
And.. in order to experiment, I stopped zigbee2mqtt, deleted the file, and started again. The file does not get re-created, but everything works fine (including Homeassistant). What is the critical function of the file and how it is treated?

Thanks for helping me out ;-)

greetings from NL, Jasper.

**Comment by ghost** [comment]
UPDATE: I posted this message, and 2 seconds afterwards I see that the state.json has been recreated. Maybe I need to be a bit more patient.

So, I still would like to understand if the state.json file:
- can be deleted like I did, without any disturbance to the functioning of the application
- is used for any other reason than for inquery

many thanks, greetings

**Comment by Koenkk** [comment]
- `state.json` is saved every 5 mins when z2m is running and when z2m is stopped
- it's safe to delete
  - When z2m is running, deleting doesn't have any influence (as the state is also kept in memory)
  - When z2m is not running, z2m doesn't know the states of devices anymore so e.g. it cannot send all the states to Home Assistant anymore. This is of course until the devices send their states again.

**Comment by ghost** [comment]
Clear answer Koen, thanks for replying!

**Comment by xDereCx** [comment]
are you sure about the deleting state.json while z2m is not running? i have lost complete setting of the valves, switches etc. everything end up just with "?" absolute mess after deleting

**Reply by andrei-lazarov** [reply]
Devices keep the values. Just Z2M doesn't know them anymore.

**Reply by xDereCx** [reply]
problem is, when Z2M needs to report current state in connection with openhab and the value is not present. example got thermostatic valves in a rooms, heating on/off scripts needs valve position to start/stop heating. but valve position is not present. also the state of the switch hot water/heating is not known...
